main/pipeline/pipeline.cc

```diff
@@ -1083,6 +1087,11 @@ bool cacheTreesAndFiles(const core::GlobalState &gs, WorkerPool &workers, vector
                     auto &file = job->file.data(gs);
                     if (!file.cached && !file.hasParseErrors) {
                         threadResult.emplace_back(fileKey(file), core::serialize::Serializer::storeFile(file, *job));
+                        // Stream out compressed files so that writes happen in parallel with processing.
+                        if (processedByThread > 100) {
+                            resultq->push(move(threadResult), processedByThread);
+                            processedByThread = 0;
+                        }
```

Comment:
I ran this change w/o streaming (w/ just the new metric) and it shows that worker time is the same before and after this change, so even if we're interrupting worker threads it's not meaningfully disrupting them.

> Does that match your understanding, or am I missing something?

? The coordinator thread definitely checks the output queue in parallel with worker threads, and it can grab work prior to the worker threads finishing.

Reply:
Oh, oh, I see: for some reason I thought the `resultq` loop was running on the worker threads.  OK, so we're pushing smaller chunks of work onto `resultq`, so the `resultq` loop on the main thread can start processing results sooner.  I see how this works.

Sorry for the confusion; thanks for working through things with me!